packages/integration-tests/suites/replay/sessionInactive/test.ts

```diff
@@ -0,0 +1,88 @@
+import { expect } from '@playwright/test';
+import type { ReplayEvent } from '@sentry/types';
+
+import { sentryTest } from '../../../utils/fixtures';
+import { envelopeRequestParser } from '../../../utils/helpers';
+import { getExpectedReplayEvent } from '../../../utils/replayEventTemplates';
+import {
+  getFullRecordingSnapshots,
+  getIncrementalRecordingSnapshots,
+  getReplaySnapshot,
+  normalize,
+  shouldSkipReplayTest,
+  waitForReplayRequest,
+} from '../../../utils/replayHelpers';
+
+// Session should expire after 2s - keep in sync with init.js
+const SESSION_TIMEOUT = 2000;
+
+sentryTest('handles an inactive session', async ({ getLocalTestPath, page }) => {
+  if (shouldSkipReplayTest()) {
+    sentryTest.skip();
+  }
+
+  const reqPromise0 = waitForReplayRequest(page, 0);
+
+  await page.route('https://dsn.ingest.sentry.io/**/*', route => {
+    return route.fulfill({
+      status: 200,
+      contentType: 'application/json',
+      body: JSON.stringify({ id: 'test-id' }),
+    });
+  });
+
+  const url = await getLocalTestPath({ testDir: __dirname });
+
+  await page.goto(url);
+
+  const replayEvent0 = envelopeRequestParser(await reqPromise0) as ReplayEvent;
+  expect(replayEvent0).toEqual(getExpectedReplayEvent({}));
+
+  const fullSnapshots0 = getFullRecordingSnapshots(await reqPromise0);
+  expect(fullSnapshots0.length).toEqual(1);
+  const stringifiedSnapshot = normalize(fullSnapshots0[0]);
+  expect(stringifiedSnapshot).toMatchSnapshot('snapshot-0.json');
+
+  await page.click('#button1');
+
+  // We wait for another segment 0
+  const reqPromise1 = waitForReplayRequest(page, 0);
+
+  // Now we wait for the session timeout, nothing should be sent in the meanwhile
+  await new Promise(resolve => setTimeout(resolve, SESSION_TIMEOUT));
+
+  // nothing happened because no activity/inactivity was detected
+  const replay = await getReplaySnapshot(page);
+  // @ts-ignore private api
+  expect(replay._isEnabled).toEqual(true);
+  // @ts-ignore private api
+  expect(replay._isPaused).toEqual(false);
+
+  // Now we trigger a blur event, which should move the session to paused mode
+  await page.evaluate(() => {
+    window.dispatchEvent(new Event('blur'));
+  });
+
+  const replay2 = await getReplaySnapshot(page);
+  // @ts-ignore private api
+  expect(replay2._isEnabled).toEqual(true);
+  // @ts-ignore private api
+  expect(replay2._isPaused).toEqual(true);
+
+  // Trigger an action, should re-start the recording
+  await page.click('#button2');
+
+  const replay3 = await getReplaySnapshot(page);
+  // @ts-ignore private api
+  expect(replay3._isEnabled).toEqual(true);
+  // @ts-ignore private api
+  expect(replay3._isPaused).toEqual(false);
+
```

Comment:
I know we already have such tests but I'm not sure if we want to add more of them: IMO we're inspecting intrinsics too much for an integration test by checking the internal state of the `ReplayContainer`. 
That being said, I get that it's hard to check that nothing happens in this situation otherwise. So my ramble is probably pointless 😅 
IIRC we already have some unit integration tests for paused replays. Any chance we could use/adjust them instead? 

But I don't want to block this too long so if there's no feasible option, let's just roll with this test instead!

Reply:
Yeah, I totally get you. I mean we do have some unitgration (xD) tests for this, so we may also skip this here - I figured this is a bit more "real" as it actually waits the correct time and we have less/no mocking/overwriting of stuff. IMHO it's "OK" to have this for now, but we should keep an eye on this so this doesn't go overboard 😅

Reply:
> unitgration

I love it 😅 

Sure, sounds fair - let's just try to minimize using this in the future whenever there's another possiblilty.